File: src/main/java/com/star/easydoc/service/generator/DocGenerator.java
package com.star.easydoc.service.generator;

import com.intellij.psi.PsiElement;

/**
 * @author wangchao
 * @date 2019/08/31
 */
public interface DocGenerator {
    /**
     * 生成
     *
     * @param psiElement psiElement
     * @return java.lang.String
     */
    String generate(PsiElement psiElement);
}

File: easydoc-kdoc/src/main/kotlin/com/star/easydoc/kdoc/service/generator/DocGenerator.kt
package com.star.easydoc.kdoc.service.generator

import com.intellij.psi.PsiElement

/**
 * 文档生成器
 *
 * @author wangchao
 * @date 2019/11/12
 */
interface DocGenerator {
    /**
     * 生成
     *
     * @param psiElement psiElement
     * @return java.lang.String
     */
    fun generate(psiElement: PsiElement): String
}